pkg/controller/operators/olm/apiservices.go

```diff
@@ -729,6 +731,19 @@ func (a *Operator) installAPIServiceRequirements(desc v1alpha1.APIServiceDescrip
 	// create a fresh CA bundle
 	apiService.Spec.CABundle = caPEM
 
+	// set apiservice owner to packageserver for proper cleanup on cluster upgrade
+	// only for packageserver api service
+	if apiService.Name == PackageserverName {
+		if l := apiService.GetLabels(); l != nil {
+			if apiService.Labels[ownerutil.OwnerKey] != ownerutil.OwnerPackageServer {
+				apiService.Labels[ownerutil.OwnerKey] = ownerutil.OwnerPackageServer
+			}
+		} else {
+			apiService.Labels = make(map[string]string)
+			apiService.Labels[ownerutil.OwnerKey] = ownerutil.OwnerPackageServer
```

Comment:
isn't the `OwnerNamespaceKey` also important?

Reply:
Yes it is, and is also handled by `ownerutil.AddOwnerLabels(...)` via `OwnerLabel()`
```
// OwnerLabel returns a label added to generated objects for later querying
func OwnerLabel(owner Owner, kind string) map[string]string {
	return map[string]string{
		OwnerKey:          owner.GetName(),
		OwnerNamespaceKey: owner.GetNamespace(),
		OwnerKind:         kind,
	}
}
```